### ISSUE TITLE
Does it support edge detection models?

### ISSUE THREAD

**Issue by kadirnar** [mert-kurttutan/torchview#44]
Hello @mert-kurttutan ,

I am developing the LDC model for edge detection. Can you help visualize the architecture?

Code: LDC Model: https://github.com/xavysp/LDC/blob/main/modelB4.py#L143
Paper: https://ieeexplore.ieee.org/stamp/stamp.jsp?arnumber=9807316
```
from torchview import draw_graph
model = LDC()
model_graph = draw_graph(model, expand_nested=True, input_size=(1,3,32,32))
model_graph.visual_graph()
```

Error Message:
```
.../python3.8/site-packages/torchview/computation_graph.py", line 124, in render_edges tail_id, head_id = self.id_dict[tail.node_id], self.id_dict[head.node_id]
KeyError: '140104182242320'
```

**Comment by mert-kurttutan** [comment]
TLDR: 
I added new branch called `empty-pass-node`. I check this with your code (in different settings) and other models that can lead to similar problems (transformers model from huggings face). In all of them, it works as expected. You can use this branch before I push into main branch (and upload the recent version to pypi)

Using `empty-pass-node` branch, I obtained the following result,

![LDC](https://user-images.githubusercontent.com/88637659/207900682-989099f0-a8b0-453f-85d2-05910b0761c1.png)

I also changed a few parameter to see if the result is robust, it still gives correct representation


More detail,
I checked the problem and it seems to stem from the reuse of output tensors inside the `forward()`
e.g.
```
forward(self, x1, x2):
  y = x1 + x2
  return x1, x2
```
where x2 is outputs and is used inside forward prop.  For instance, this happens in `_denselayer` module. To deal with this, I introduced empty pass forward that redirected reused output tensors.

I will prepare tests cases to make sure that this problem is indeed solved.
You are also welcome to give possible test case that you think is valuable

**Comment by kadirnar** [comment]
You are awesome. This is very nice. 
https://github.com/kadirnar/Pytorch-LDC/blob/main/backbone/ghostnet_edge.py#L79-L84

I updated the codes with ghostnet architecture. It is working now. This library is great. But can we save it as .html or .png file?

```
 from torchview import draw_graph
 model = LDC()
 model_graph = draw_graph(model, input_size=(1, 3, 352, 352), expand_nested=True)
 model_graph.visual_graph.view()
```
It just saves as a .pdf file.

**Comment by mert-kurttutan** [comment]
For graph, you have 2 options (which do the same thing)

1. Set the `save_graph=True` when calling `draw_graph`
or
2. Use the result graph: 
```
model_graph = draw_graph(...)
model_graph.visual_graph.render(format='png')
```
Format specifies the image format, 'pdf', 'png' etc
Note: There is also directory argument in `draw_graph` if you want to change the location to save the image.

**Comment by kadirnar** [comment]
Thank you. It worked for me.
